File: CS/NET40/WhatIsMyBrowser.CommonTypes/SoftwareStream.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace WhatIsMyBrowser.CommonTypes
{
    public class SoftwareStream
    {
        [JsonProperty("latest_version")]
        public int[] LatestVersion { get; set; }

        [JsonProperty("download_url")]
        public Uri DownloadUrl { get; set; }

        [JsonProperty("update")]
        public string Update { get; set; }

        [JsonProperty("update_url")]
        public Uri UpdateUrl { get; set; }

        [JsonProperty("release_date")]
        public string ReleaseDate { get; set; }

        [JsonProperty("package_type")]
        public string PackageType { get; set; }

        [JsonProperty("sample_user_agents")]
        public Dictionary<string, string[]> SampleUserAgents { get; set; }
    }
}

File: VB/NET40/WhatIsMyBrowser.CommonTypesVB/SoftwareStream.vb
Imports Newtonsoft.Json

Public Class SoftwareStream
    <JsonProperty("latest_version")>
    Public Property LatestVersion As Integer()

    <JsonProperty("download_url")>
    Public Property DownloadUrl As Uri

    <JsonProperty("update")>
    Public Property Update As String

    <JsonProperty("update_url")>
    Public Property UpdateUrl As Uri

    <JsonProperty("release_date")>
    Public Property ReleaseDate As String

    <JsonProperty("package_type")>
    Public Property PackageType As String

    <JsonProperty("sample_user_agents")>
    Public Property SampleUserAgents As Dictionary(Of String, String())
End Class
